core/io/resource.cpp

```diff
@@ -37,6 +37,11 @@
 #include "core/os/os.h"
 #include "scene/main/node.h" //only so casting works
 
+#ifdef TOOLS_ENABLED
+#include "editor/editor_node.h"
+#include "editor/editor_string_names.h"
+#endif // TOOLS_ENABLED
```

Comment:
Haven't reviewed details of that PR at all, but just flagging this: `core` code shouldn't depend on `editor` code, that's breaking encapsulation rules: https://github.com/godotengine/godot/issues/53295

Reply:
Thanks for letting me know!
I've fixed this particular instance by inverting the coupling, via an overridable function pointer, similar to what's done in other places in the engine.
It also created a great home for the macros, so this helped in multiple ways!